### PR TITLE
Add Object Observation Station (OOS)

### PR DESCRIPTION
**Pull request by MuXiu1997**
Detect adjacent blocks to complete Retrieval Task / Fluid Task

It is still possible to modify textures

https://user-images.githubusercontent.com/49554020/169690781-09eb315a-91f1-4054-a243-1ccd19f8a4c0.mp4

### CODE REVIEW

File: src/main/java/betterquesting/blocks/TileObservationStation.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        } catch (Exception ignored) {
+            this.owner = null;
+        }
+    }
+
+    @Override
+    public void writeToNBT(@Nonnull NBTTagCompound nbt) {
+        super.writeToNBT(nbt);
+        if (this.owner == null) return;
+        nbt.setString("owner", this.owner.toString());
```
Comment (miozune) [inline src/main/java/betterquesting/blocks/TileObservationStation.java:134]: `NBT_KEY_OWNER`

File: src/main/java/betterquesting/blocks/TileObservationStation.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            if (player.getGameProfile().getId().equals(uuid)) return player;
+        }
+
+        return null;
+    }
+
+    @Override
+    public void readFromNBT(@Nonnull NBTTagCompound nbt) {
+        super.readFromNBT(nbt);
+        if (!nbt.hasKey(NBT_KEY_OWNER, 8)) return;
```
Comment (miozune) [inline src/main/java/betterquesting/blocks/TileObservationStation.java:122]: Use `net.minecraftforge.common.util.Constants.NBT` for readability

**Review by miozune — COMMENTED** [review]
Can you add `.DS_Store` to `.gitignore`?

**Comment by MuXiu1997** [comment]
> Can you add `.DS_Store` to `.gitignore`?

Didn't realise it wasn't in there

File: src/main/java/betterquesting/api/questing/tasks/IItemTask.java [hunk 3]
```diff
@@ -10,4 +11,6 @@ public interface IItemTask extends ITask
 {
 	boolean canAcceptItem(UUID owner, DBEntry<IQuest> quest, ItemStack stack);
 	ItemStack submitItem(UUID owner, DBEntry<IQuest> quest, ItemStack stack);
+
+    default void retrievalItems(ParticipantInfo pInfo, DBEntry<IQuest> quest, ItemStack[] items) {}
```
Comment (miozune) [inline src/main/java/betterquesting/api/questing/tasks/IItemTask.java:15]: `retrieval` is noun, `retrieve` is better imao

File: src/main/java/betterquesting/blocks/TileObservationStation.java (hunk 2 above)
Comment (MuXiu1997) [reply src/main/java/betterquesting/blocks/TileObservationStation.java:122]: good to know

File: src/main/java/betterquesting/api/questing/tasks/IItemTask.java (hunk 3 above)
Comment (MuXiu1997) [reply src/main/java/betterquesting/api/questing/tasks/IItemTask.java]: done

File: src/main/java/betterquesting/blocks/TileObservationStation.java (hunk 1 above)
Comment (MuXiu1997) [reply src/main/java/betterquesting/blocks/TileObservationStation.java]: done

File: src/main/java/betterquesting/blocks/TileObservationStation.java (hunk 2 above)
Comment (MuXiu1997) [reply src/main/java/betterquesting/blocks/TileObservationStation.java]: done

**Review by miozune — APPROVED** [review]

**Comment by MuXiu1997** [comment]
> Do you have to select quest u want to complete or it will check all of them

All active quests

**Pull request merged**